clouduniqueid/clouds/gcp/__init__.py

```diff
@@ -19,63 +19,86 @@ def check_missing_data_keys(dataKeys: list, uniqueIdFormat: str):
-
         uniqueIds: Dict = unique_ids
-        data = {k.lower().replace("_", ""): v for k, v in data .items()}
+        data = {k.lower().replace("_", ""): v for k, v in data.items()}
         dataKeys = list(data.keys())
 
         if not uniqueIds.get(service, None):
             logger.error(f"GCP service {service} unknown")
             raise ValueError(f"GCP service {service} unknown")
 
         elif not uniqueIds.get(service, {}).get(resourceType, None):
-            logger.error(f"GCP service {service} resource type {resourceType} not supported")
-            raise ValueError(f"GCP service {service} resource type {resourceType} not supported")
+            logger.error(
+                f"GCP service {service} resource type {resourceType} not supported",
+            )
+            raise ValueError(
+                f"GCP service {service} resource type {resourceType} not supported",
+            )
 
-        elif 'projectId' in uniqueIds[service][resourceType] and not projectId:
-            uniqueIdFormat = uniqueIds[service][resourceType].replace(" ", "").\
-                replace('data["', "").replace('".lower()]', "")
+        elif "projectId" in uniqueIds[service][resourceType] and not projectId:
+            uniqueIdFormat = (
+                uniqueIds[service][resourceType]
+                .replace(" ", "")
+                .replace('data["', "")
+                .replace('".lower()]', "")
+            )
             logger.error("GCP projectId required")
             raise ValueError(f"Invalid parameters provided, GCP projectId required,\
                 uniqueId format for resource {service} {resourceType} is {uniqueIdFormat}")
 
-        elif 'location' in uniqueIds[service][resourceType] and not location:
-            uniqueIdFormat = uniqueIds[service][resourceType].replace(" ", "").replace('data["', "").\
-                replace('".lower()]', "")
+        elif "location" in uniqueIds[service][resourceType] and not location:
+            uniqueIdFormat = (
+                uniqueIds[service][resourceType]
+                .replace(" ", "")
+                .replace('data["', "")
+                .replace('".lower()]', "")
+            )
             logger.error("GCP location required")
             raise ValueError(
                 f"Invalid parameters provided, GCP location required, ,\
                     uniqueId format for resource {service} {resourceType} is {uniqueIdFormat}",
             )
 
         else:
-            uniqueIdFormat = uniqueIds[service][resourceType].replace(" ", "").\
-                replace('data["', "").replace('".lower()]', "")
+            uniqueIdFormat = (
+                uniqueIds[service][resourceType]
+                .replace(" ", "")
+                .replace('data["', "")
+                .replace('".lower()]', "")
+            )
             missingKeys = check_missing_data_keys(dataKeys, uniqueIdFormat)
             if len(missingKeys) > 0:
-                errorMsg = ''
```

Comment:
<b>Vulnerability:</b> Detected the use of eval(). eval() can be dangerous if used to evaluate dynamic content. If this content can be input from outside the program, this may be a code injection vulnerability. Ensure evaluated content is not definable by external sources.<br/>
                            <b>Lines</b>: <b>Lines</b>: 114 to 114<br/>
                            <b>OWASP:</b> A03:2021 - Injection<br/>
                            <b><a href="https://console.cloudanix.com/code?t=vulnerabilities" target='_blank'>Click here for more details</a></b>